compiler/rustc_ast_passes/src/errors.rs

```diff
@@ -823,3 +823,10 @@ pub(crate) struct DuplicatePreciseCapturing {
     #[label]
     pub bound2: Span,
 }
+
+#[derive(Diagnostic)]
+#[diag(ast_passes_missing_extern_abi)]
+pub(crate) struct MissingExternABI {
```

Comment:
```suggestion
#[diag(ast_passes_extern_without_abi)]
pub(crate) struct MissingAbi {
```

Actually, we should probably name these like this, to match the names from the corresponding lint.